### DISCUSSION TITLE
`ls-remote` for non-GitHub based tools

### DISCUSSION THREAD

**Discussion by sassdavid** [jdx/mise#3337] [General]
Hi,

I'm writing about a previous issue: https://github.com/jdx/mise/issues/3050

I'm not entirely familiar with how originally `aqua` works, so I'd like to clarify something.

Am I correct in understanding that there's no possibility to use `ls-remote` for tools that don't have tags/branches on GitHub, or is `ls-remote` completely unavailable for all aqua-provided tools?

For example, is running `mise ls-remote helm` equivalent to `mise ls-remote aqua:helm/helm`? Or is there a possibility that it uses `asdf` in the background to list remote versions?

If my first assumption is correct, wouldn't it be possible to provide an option to query remote versions of such tools?

To use the `1password` example, its `asdf` plugin uses the following command to fetch remote versions:
```
curl -s https://app-updates.agilebits.com/product_history/CLI2 | sed -n '/<h3/{n;p;}' | sed 's/[[:space:]]//g'
```
(Of course I am aware that this is not the best solution.)

Could we consider adding additional configurations for tools in the https://github.com/jdx/mise/blob/5c4332eccb5f807d410386c05700d67a3df3fc6d/registry.toml file? Would this help resolve the `ls-remote` issues for tools that don't have a GitHub repository at all?

**Comment by jdx** [comment]
this is a question for aqua. we should be using whatever version fetching it uses, not building our own strategy on top of it
